test/functional/terminal/helpers.lua

```diff
@@ -4,19 +4,28 @@
 local helpers = require('test.functional.helpers')(nil)
 local Screen = require('test.functional.ui.screen')
 local testprg = helpers.testprg
+local exec_lua = helpers.exec_lua
 local feed_command, nvim = helpers.feed_command, helpers.nvim
 
 local function feed_data(data)
-  -- A string containing NUL bytes is not converted to a Blob when
-  -- calling nvim_set_var() API, so convert it using Lua instead.
-  nvim('exec_lua', 'vim.g.term_data = ...', {data})
-  nvim('command', 'call jobsend(b:terminal_job_id, term_data)')
+  exec_lua('vim.api.nvim_chan_send(vim.b.terminal_job_id, ...)', data)
 end
 
 local function feed_termcode(data)
-  -- feed with the job API
-  nvim('command', 'call jobsend(b:terminal_job_id, "\\x1b'..data..'")')
+  feed_data('\x1b' .. data)
```

Comment:
This does not work with Lua 5.1. Use `'\027'` instead.
```suggestion
  feed_data('\027' .. data)
```